go-controller/pkg/libovsdbops/switch.go

```diff
@@ -201,33 +201,48 @@ func AddACLsToLogicalSwitchOps(nbClient libovsdbclient.Client, ops []libovsdb.Op
 // corresponding ops
 func RemoveACLsFromLogicalSwitchesWithPredicateOps(nbClient libovsdbclient.Client, ops []libovsdb.Operation,
 	p switchPredicate, acls ...*nbdb.ACL) ([]libovsdb.Operation, error) {
-	sw := nbdb.LogicalSwitch{
-		ACLs: make([]string, 0, len(acls)),
-	}
+	aclUUIDs := make(map[string]bool, len(acls))
 	for _, acl := range acls {
-		sw.ACLs = append(sw.ACLs, acl.UUID)
+		aclUUIDs[acl.UUID] = true
 	}
-	opModel := operationModel{
-		Model:            &sw,
-		ModelPredicate:   p,
-		OnModelMutations: []interface{}{&sw.ACLs},
-		ErrNotFound:      false,
-		BulkOp:           true,
+	switches, err := FindLogicalSwitchesWithPredicate(nbClient, p)
+	if err != nil {
+		return ops, fmt.Errorf("failed to remove acls from the logical switches with predicate, "+
+			"predicate search failed: %w", err)
+	}
+	opModels := make([]operationModel, 0, len(switches)+1)
+	for _, sw := range switches {
+		sw := sw
+		aclsToRemove := []string{}
+		for _, swACLUUID := range sw.ACLs {
+			if aclUUIDs[swACLUUID] {
+				aclsToRemove = append(aclsToRemove, swACLUUID)
+			}
+		}
+		sw.ACLs = aclsToRemove
+		opModel := operationModel{
+			Model:            sw,
+			OnModelMutations: []interface{}{&sw.ACLs},
+			ErrNotFound:      false,
+			BulkOp:           true,
+		}
+		opModels = append(opModels, opModel)
 	}
-
 	m := newModelClient(nbClient)
-	return m.DeleteOps(ops, opModel)
+	return m.DeleteOps(ops, opModels...)
 }
 
 // RemoveACLsFromLogicalSwitchesWithPredicate looks up logical switches from the cache
 // based on a given predicate and removes from them the provided ACLs
 func RemoveACLsFromLogicalSwitchesWithPredicate(nbClient libovsdbclient.Client, p switchPredicate, acls ...*nbdb.ACL) error {
-	ops, err := RemoveACLsFromLogicalSwitchesWithPredicateOps(nbClient, nil, p, acls...)
-	if err != nil {
+	return batch[*nbdb.ACL](defaultBatchSize, acls, func(batchACLs []*nbdb.ACL) error {
```

Comment:
I think batching should happen outside of libovsdbops for now given that is contextual and not applied to all ops basically because from within libovsdbops we don't have the information we need to consistently apply batching.

Reply:
For `RemoveACLsFromLogicalSwitchesWithPredicate` we know that it is safe to apply batching right? I also think if we need to use this function on another sync, for some other stale acls in the future, it would be nice to re-use batching and not copy it?

Reply:
Yes, but we might not need it in all future uses of `RemoveACLsFromLogicalSwitchesWithPredicate`. I agree it does no harm, but what about other ops that might be also subject to batching like `RemoveACLsFromLogicalSwitchesWithPredicateOps`, `DeleteNATs`, `DeleteAddressSets`.... For some of those, we might have a plan to add batching within libovsdbops over time but until then it is a question that people will ask themselves if it is needed or not. For others, like `..Ops` versions of the methods, we won't be able to add batching here, so if it is needed, it will end up elsewhere. So at the end, looking forward, it won't be consistent.
So for now, I think it is better to apply the batching elsewhere with a comment reasoning why was that needed.
The `batch` utility already provides a level of code re-use and certainly there are other possibilities to share the code.
This is a nit but clearly for me adding batching within libovsdbops is not the best option if it can't be consistently applied to all ops going forward. If you insist, I guess we can leave it here

Reply:
I like that reasoning! thanks for sharing your thoughts on that